apps/spruce/src/components/Spawn/utils/hostUptime.ts

```diff
@@ -0,0 +1,247 @@
+    : enabledWeekdaysCount * getDailyUptime({ startTime, stopTime });
+  return { enabledHoursCount, enabledWeekdaysCount };
+};
+
+export const getSleepSchedule = (
+  { sleepSchedule, useDefaultUptimeSchedule }: HostUptime,
+  timeZone: string,
+): SleepScheduleInput => {
+  if (useDefaultUptimeSchedule) {
+    return {
+      ...defaultSleepSchedule,
+      timeZone,
+    };
+  }
+
+  const {
+    enabledWeekdays,
+    timeSelection: { runContinuously, startTime, stopTime },
+  } = sleepSchedule;
+
+  return {
+    dailyStartTime: runContinuously ? "" : toTimeString(new Date(startTime)),
+    dailyStopTime: runContinuously ? "" : toTimeString(new Date(stopTime)),
+    permanentlyExempt: false,
+    temporarilyExemptUntil: null,
+    timeZone,
+    shouldKeepOff: false,
+    wholeWeekdaysOff: enabledWeekdays.reduce((accum, isEnabled, i) => {
+      if (!isEnabled) {
+        accum.push(i);
+      }
+      return accum;
+    }, []),
+  };
+};
+
+const getDailyUptime = ({ startTime, stopTime }) =>
+  differenceInHours(new Date(stopTime), new Date(startTime));
+
+const toTimeString = (date: Date): string =>
+  date.toLocaleTimeString([], {
+    hour: "2-digit",
+    hour12: false,
+    minute: "2-digit",
+  });
+
+export const defaultSleepSchedule: Omit<SleepScheduleInput, "timeZone"> = {
+  dailyStartTime: toTimeString(defaultStartDate),
+  dailyStopTime: toTimeString(defaultStopDate),
+  permanentlyExempt: false,
+  // TODO: Add pause
+  shouldKeepOff: false,
+  temporarilyExemptUntil: null,
+  wholeWeekdaysOff: [0, 6],
+};
+
+export const getHostUptimeFromGql = (
+  sleepSchedule: MyHost["sleepSchedule"],
+): HostUptime => {
+  if (!sleepSchedule) return null;
+
+  if (matchesDefaultUptimeSchedule(sleepSchedule)) {
+    return {
+      useDefaultUptimeSchedule: true,
+    };
+  }
+
+  const { dailyStartTime, dailyStopTime, wholeWeekdaysOff } = sleepSchedule;
+
+  return {
+    useDefaultUptimeSchedule: false,
+    sleepSchedule: {
+      enabledWeekdays: new Array(7)
+        .fill(false)
+        .map((_, i) => !wholeWeekdaysOff.includes(i)),
+      timeSelection:
+        dailyStartTime && dailyStopTime
+          ? {
+              startTime: parse(dailyStartTime, "HH:mm", new Date()).toString(),
+              stopTime: parse(dailyStopTime, "HH:mm", new Date()).toString(),
+              runContinuously: false,
+            }
+          : {
+              startTime: defaultStartDate.toString(),
+              stopTime: defaultStopDate.toString(),
+              runContinuously: true,
+            },
+    },
+  };
+};
+
+const matchesDefaultUptimeSchedule = (
+  sleepSchedule: MyHost["sleepSchedule"],
+): boolean => {
+  const { dailyStartTime, dailyStopTime, wholeWeekdaysOff } = sleepSchedule;
+
+  if (
+    [...wholeWeekdaysOff].sort().toString() !== new Array([0, 6]).toString()
+  ) {
+    return false;
+  }
+  if (dailyStartTime !== "08:00") return false;
+  if (dailyStopTime !== "20:00") return false;
```

Comment:
I think it could be nice to avoid hardcoding this value! Is there something we can reuse from the above `const`?